src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -373,15 +378,16 @@ public void fireFileStarted(String fileName) {
     /**
      * Notify all listeners about the errors in a file.
      *
-     * @param fileName the audited file
-     * @param errors the audit errors from the file
+     * @param fileProcessingResult file processing results object
      */
     @Override
-    public void fireErrors(String fileName, SortedSet<LocalizedMessage> errors) {
+    public void fireErrors(FileProcessingResult fileProcessingResult) {
```

Comment:
`fileProcessingResult.getFileText` is not used it is already sign smth is wrong in design.

please see my comment at https://github.com/checkstyle/checkstyle/issues/3573#issuecomment-314760202